go-controller/pkg/kube/annotator.go

```diff
@@ -0,0 +1,102 @@
+package kube
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	kapi "k8s.io/api/core/v1"
+)
+
+// Annotator represents the exported methods for handling node annotations
+type Annotator interface {
+	Set(key string, value interface{}) error
+	SetWithFailureHandler(key string, value interface{}, failFn FailureHandlerFn) error
+	Del(key string)
```

Comment:
we don't normally abbreviate "Delete" so it makes it harder to remember the API if you do it here